server/huntflow_reloaded/handler.py

```diff
@@ -298,3 +299,197 @@ def get_date_from_string(date_string):
         .replace(tzinfo=None)
 
     return data
+
+
+class TokenObtainPairHandler(RequestHandler): # pylint: disable=abstract-method,
+    """Class implementing obtaining tokens pair. """
+
+    def __init__(self, application, request, **kwargs):
+        super(TokenObtainPairHandler, self).__init__(application, request,
+                                                     **kwargs)
+        self.user = None
+        self.valid = False
+
+    def initialize(self, postgres_url):  # pylint: disable=arguments-differ
+        self._postgres_url = postgres_url
+
+    async def post(self):  # pylint: disable=arguments-differ
+        try:
+            email = self.get_argument('email')
+            password = self.get_argument('password')
+        except KeyError:
+            raise IncompleteRequest
+
+        await self.validate(email, password)
+
+        if self.valid:
+            refresh = RefreshToken.for_user(self.user.id)
+
+            data = {
+                'access': str(refresh.access_token()), 'refresh': str(refresh)}
+        else:
+            data = {
+                "detail": "No active account found with the given credentials"}
+            self.set_status(400)
+
+        self.write(data)
+
+    async def validate(self, email, password):
+        """Checks if user with the given credentials exists. """
+
+        await models.gino_run(self._postgres_url)
+
+        user = await models.User.query.where(
+            models.User.email == email).gino.first()
+
+        if user and user.password == password:
+            self.user = user
+            self.valid = True
+
+
+class TokenRefreshHandler(RequestHandler):  # pylint: disable=abstract-method
+    """Class implementing handler for refreshing tokens pair request."""
+
+    async def post(self):  # pylint: disable=arguments-differ
+        try:
+            refresh = RefreshToken(self.get_argument('refresh'))
+        except ExpiredTokenException:
+            self.set_status(401)
+            data = {"detail": "Refresh token is expired"}
+        else:
+            data = {'access': str(refresh.access_token())}
+
+        self.write(data)
+
+
+class ManageHandler(RequestHandler): # pylint: disable=abstract-method,
+    """Class implementing abstraction for /manage endpoint. """
+
+    async def _connect_to_database(self):
+        """ Connecting to ORM """
+
+        await models.gino_run(self._postgres_url)  # pylint: disable=no-member
+
+    def get_current_user(self):
+        try:
+            access_token = self.get_argument('access')
+        except MissingArgumentError:
+            self.write({"detail": "Token is not provided"})
+            self.set_status(500)
```

Comment:
I guess in this case we should return `400 Bad Request` but definitely not `500 Internal Server Error`.